staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "apiserver"
+	subsystem = "flowcontrol"
+)
+
+const (
+	priorityLevel = "priorityLevel"
+	flowSchema    = "flowSchema"
+)
+
+var (
+	queueLengthBuckets            = []float64{0, 10, 25, 50, 100, 250, 500, 1000}
+	requestDurationSecondsBuckets = []float64{0, 0.005, 0.02, 0.05, 0.1, 0.2, 0.5, 1, 2, 5, 10, 30}
+)
+
+func init() {
+	prometheus.MustRegister(apiserver_rejected_requests)
+	prometheus.MustRegister(apiserver_current_inqueue_requests)
+	prometheus.MustRegister(apiserver_request_queue_length)
+	prometheus.MustRegister(apiserver_request_concurrency_limit)
+	prometheus.MustRegister(apiserver_current_executing_requests)
+	prometheus.MustRegister(apiserver_requests_waiting_seconds)
+	prometheus.MustRegister(apiserver_requests_execution_seconds)
+}
+
+var (
+	apiserver_rejected_requests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_rejected_requests",
+			Help:      "Number of rejected requests by request-management system",
+		},
+		[]string{priorityLevel, "reason"},
+	)
+	apiserver_current_inqueue_requests = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_current_inqueue_requests",
+			Help:      "Number of requests currently pending in the queue by request-management system",
+		},
+		[]string{priorityLevel},
+	)
+	apiserver_request_queue_length = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_request_queue_length",
+			Help:      "Length of queue in the request-management system",
+			Buckets:   queueLengthBuckets,
+		},
+		[]string{priorityLevel},
+	)
+	apiserver_request_concurrency_limit = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "apiserver_request_concurrency_limit",
+			Help:      "Limit of shared concurrency in the request-management system",
```

Comment:
I think including the word "shared" here is potentially confusing.  It is not clear to the reader what is being shared with what.  Since this metric has a label identifying a scope, if we strike the word "shared" from here then the natural interpretation will be that this is the concurrency limit for the scope of the metric --- which is what we want to say.

Also, I think we can shorten "Limit of concurrency" to "concurrency limit".

Reply:
renamed

Reply:
I still see "shared" in there.  I think this word is misleading.  We used the term "shared" when we had multiple priority levels making dynamic deductions from a shared pool of capacity.  But since then we have simplified that away.

I think better terminology would be simply "concurrency limit" or perhaps "priority level's concurrency limit" or "concurrency limit of priority level" or "concurrency limit by priority level".